test/test_ctparser_utils.jl

```diff
@@ -36,20 +36,25 @@ e = :( ∫( x[1](t)^2 + 2*u(t) ) → min )
 @test !has(e, :v, :t)
 
 t = :t; t0 = 0; tf = :tf; x = :x; u = :u
+@test constraint_type(:( x[1:2:5](0)          ), t, t0, tf, x, u) == (:initial, 1:2:5)
 @test constraint_type(:( x[1:2](0)          ), t, t0, tf, x, u) == (:initial, 1:2)
 @test constraint_type(:( x[1](0)            ), t, t0, tf, x, u) == (:initial, Index(1))
 @test constraint_type(:( 2x[1](0)^2         ), t, t0, tf, x, u) == (:boundary, :(2 * var"x#0"[1] ^ 2))
+@test constraint_type(:( x[1:2:5](tf)         ), t, t0, tf, x, u) == (:final, 1:2:5)
 @test constraint_type(:( x[1:2](tf)         ), t, t0, tf, x, u) == (:final, 1:2)
 @test constraint_type(:( x[1](tf)           ), t, t0, tf, x, u) == (:final, Index(1))
 @test constraint_type(:( x[1](tf) - x[2](0) ), t, t0, tf, x, u) == (:boundary, :(var"x#f"[1] - var"x#0"[2]))
 @test constraint_type(:( 2x[1](tf)^2        ), t, t0, tf, x, u) == (:boundary, :(2 * var"x#f"[1] ^ 2))
+@test constraint_type(:( u[1:2:5](t)          ), t, t0, tf, x, u) == (:control_range, 1:2:5)
 @test constraint_type(:( u[1:2](t)          ), t, t0, tf, x, u) == (:control_range, 1:2)
 @test constraint_type(:( u[1](t)            ), t, t0, tf, x, u) == (:control_range, Index(1))
 @test constraint_type(:( 2u[1](t)^2         ), t, t0, tf, x, u) == (:control_fun, :(2 * u[1] ^ 2))
+@test constraint_type(:( x[1:2:5](t)          ), t, t0, tf, x, u) == (:state_range, 1:2:5)
```

Comment:
@BaptisteCbl well done 👍🏽👍🏽👍🏽 after a rapid check (grep !) I agree that only changes should occur in `ctparser_utils` and `model`